addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1539,4 +1540,105 @@ options.registry.anchorName = options.Class.extend({
         this.$target.trigger('content_changed');
     },
 });
+
+WysiwygMultizone.include({
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _saveElement: function (outerHTML, recordInfo, editable) {
+        var defs = [this._super.apply(this, arguments)];
+        if ($(editable).data('oe-field') === 'mega_menu_content') {
+            var classesArray = $(editable).attr('class').toString().split(' ');
+            // Remove editable classes from editor before saving classes
+            classesArray.pop('o_editable');
+            classesArray.pop('o_dirty');
```

Comment:
Dangerous, use a whitelist instead of a blacklist, what if we introduce a new editor class at some point? We would have to remember about your feature that only 3-4 people know about. Also, maybe it would be better to do it another way: save the classes that are to be saved in the dataset of the element (so on the snippet option do: "if this class is added, then I save the class with `.data('megaMenuClasses', classes)`. That way, no need of blacklist or whitelist, just read the dataset here.